programs/compressor/Compressor.cpp

```diff
@@ -58,34 +61,40 @@ void checkAndWriteHeader(DB::ReadBuffer & in, DB::WriteBuffer & out)
 
 }
 
-#pragma GCC diagnostic ignored "-Wunused-function"
-#pragma GCC diagnostic ignored "-Wmissing-declarations"
-
 int mainEntryClickHouseCompressor(int argc, char ** argv)
 {
     using namespace DB;
+    namespace po = boost::program_options;
 
-    boost::program_options::options_description desc = createOptionsDescription("Allowed options", getTerminalWidth());
+    po::options_description desc = createOptionsDescription("Allowed options", getTerminalWidth());
     desc.add_options()
         ("help,h", "produce help message")
+        ("input", po::value<std::string>()->value_name("INPUT"), "input file")
+        ("output", po::value<std::string>()->value_name("OUTPUT"), "output file")
         ("decompress,d", "decompress")
-        ("block-size,b", boost::program_options::value<unsigned>()->default_value(DBMS_DEFAULT_BUFFER_SIZE), "compress in blocks of specified size")
+        ("offset-in-compressed-file", po::value<size_t>()->default_value(0ULL), "offset to the compressed block (i.e. physical file offset)")
+        ("offset-in-decompressed-block", po::value<size_t>()->default_value(0ULL), "offset to the decompressed block (i.e. virtual offset)")
+        ("block-size,b", po::value<unsigned>()->default_value(DBMS_DEFAULT_BUFFER_SIZE), "compress in blocks of specified size")
         ("hc", "use LZ4HC instead of LZ4")
         ("zstd", "use ZSTD instead of LZ4")
-        ("codec", boost::program_options::value<std::vector<std::string>>()->multitoken(), "use codecs combination instead of LZ4")
-        ("level", boost::program_options::value<int>(), "compression level for codecs specified via flags")
+        ("codec", po::value<std::vector<std::string>>()->multitoken(), "use codecs combination instead of LZ4")
+        ("level", po::value<int>(), "compression level for codecs specified via flags")
         ("none", "use no compression instead of LZ4")
         ("stat", "print block statistics of compressed data")
     ;
 
-    boost::program_options::variables_map options;
-    boost::program_options::store(boost::program_options::parse_command_line(argc, argv, desc), options);
+    po::positional_options_description positional_desc;
+    positional_desc.add("input", 1);
+    positional_desc.add("output", 1);
+
+    po::variables_map options;
+    po::store(po::command_line_parser(argc, argv).options(desc).positional(positional_desc).run(), options);
 
     if (options.count("help"))
     {
-        std::cout << "Usage: " << argv[0] << " [options] < in > out" << std::endl;
+        std::cout << "Usage: " << argv[0] << " [options] INPUT OUTPUT" << std::endl;
```

Comment:
I hope stdin/stdout still supported?
Note: you can seek through stdin if it is a seekable fd.

Reply:
Let's add both examples.

Reply:
>I hope stdin/stdout still supported?

Yes (But I guess you already figure this out from the test)

>Note: you can seek through stdin if it is a seekable fd.

Indeed, this is just to make this explicit